modin/engines/base/io/text/text_file_reader.py

```diff
@@ -81,3 +51,223 @@ def pathlib_or_pypath(cls, filepath_or_buffer):
         except ImportError:  # pragma: no cover
             pass
         return False
+
+    @classmethod
+    def offset(
+        cls,
+        f,
+        nrows=None,
+        skiprows=None,
+        chunk_size_bytes=None,
+        quotechar=b'"',
+        is_quoting=True,
+    ):
+        """
+        Moves the file offset at the specified amount of bytes/rows.
+
+        Parameters
+        ----------
+            f: file object
+            nrows: int, number of rows to read. Optional, if not specified will only
+                consider `chunk_size_bytes` parameter.
+            chunk_size_bytes: int, Will read new rows while file pointer
+                is less than `chunk_size_bytes`. Optional, if not specified will only
+                consider `nrows` parameter, if both not specified will read till
+                the end of the file.
+            skiprows: array or callable (optional), specifies rows to skip
+            quotechar: char that indicates quote in a file
+                (optional, by default it's '\"')
+            is_quoting: bool, Whether or not to consider quotes
+                (optional, by default it's `True`)
+
+        Returns
+        -------
+            bool: If file pointer reached the end of the file, but did not find
+            closing quote returns `False`. `True` in any other case.
+        """
+        if nrows is not None or skiprows is not None:
+            return cls._read_rows(
+                f,
+                nrows=nrows,
+                skiprows=skiprows,
+                quotechar=quotechar,
+                is_quoting=is_quoting,
+                max_bytes=chunk_size_bytes,
+            )[0]
+
+        if chunk_size_bytes is None:
+            chunk_size_bytes = -1
+
+        outside_quotes = True
+
+        chunk = f.read(chunk_size_bytes)
+        line = f.readline()  # Ensure we read up to a newline
+        # We need to ensure that one row isn't being split across different partitions
+
+        if is_quoting:
+            outside_quotes = not ((chunk.count(quotechar) + line.count(quotechar)) % 2)
+            while not outside_quotes:
+                line = f.readline()
+                outside_quotes = line.count(quotechar) % 2
+                if not line:
+                    break
```

Comment:
This is not tested - see Codecov report.

What if next situation:
csv file:
```
gfdgh, djdj, dfjdj,
1, 2, "fgjgf
dgjdj", 2, 7
4, 5, 6
```
Where first two rows - first partition.

When we read the next line, your code will show that we are not outside the quotes, and this is wrong.

Reply:
Let's say that we readed first 2 lines in a chunk, 
`chunk.count(quotechar) % 2 == 1` and then `outside_quotes == False`
Then we're going to a `while` loop and reading next line that contains one quote, so
`line.count(quotechar) % 2 == 1` and then `outside_quotes == True` and we're going out of the loop and has correct value at `outside_quotes`.

Thanks for notice that these lines are not tested, I'll add your suggested test in the next commit

Reply:
Got it